pyro/distributions/multivariate_normal.py

```diff
@@ -58,8 +58,9 @@ def rsample(self, sample_shape=torch.Size()):
 
     def log_prob(self, value):
         delta = value - self.loc
+        if len(delta.shape) == 1:
+            delta = delta.unsqueeze(-1)
         sigma_inverse = _matrix_inverse_compat(self.covariance_matrix, self.scale_tril)
-        normalization_const = ((0.5 * self.event_shape[-1]) * math.log(2 * math.pi) +
-                               self.scale_tril.diag().log().sum(-1))
-        mahalanobis_squared = (delta * torch.matmul(delta, sigma_inverse)).sum(-1)
+        normalization_const = (0.5 * self.event_shape[-1] * math.log(2 * math.pi) + self.scale_tril.diag().log().sum())
+        mahalanobis_squared = (delta.t() @ sigma_inverse @ delta).sum()
```

Comment:
@tobyclh The implementation of `log_prob` should follow Pytorch/Pyro's `batch_shape`, `event_shape` semantic. In MultivariateNormal distribution, `event_shape` is the last dimension of `loc` (number of samples in GP). We should not change this implementation. It is better to change the observation `y` to match this semantic.